### PR TITLE
Dm 1495 spdlog 2

### PR DESCRIPTION
**Pull request by mchorazak**
### Issue
[__DM-1495__](https://jira.esss.lu.se/browse/DM-1495?filter=-1)

Closes #266

*previously
[DM-1168](https://jira.esss.lu.se/browse/DM-1168)*

### Description of work
Forwarder logger now works similarly to the one in the filewriter. Logging to console, optionally logging to file and/or graylog.
- using spdlog with graylog sing plugin.
- Because graylog is not available on windows, windows conanfile has simply vanilla spdlog package.
- Logging level changed: 0 to 5, trace to critical respectively. 

### Nominate for Group Code Review

- [ ] Nominate for code review

### CODE REVIEW

File: src/EpicsClient/EpicsClientMonitor.cpp [hunk 1]
```diff
@@ -124,18 +124,12 @@ class EpicsClientMonitor_impl {
 #if TEST_PROVOKE_ERROR == 1
     static std::atomic<int> c1{0};
     if (c1 > 10) {
-      epics_client->error_in_epics();
+      epics_client->erroror_in_epics();
```
Comment (matthew-d-jones) [inline src/EpicsClient/EpicsClientMonitor.cpp]: "erroror" typo

**Comment by matthew-d-jones** [comment]
Removed Jonas as reviewer as he is on leave

File: src/EpicsClient/EpicsClientMonitor.cpp [hunk 2]
```diff
@@ -220,64 +217,60 @@ std::string ChannelRequester::getRequesterName() { return "ChannelRequester"; }
...
 }
 
 void ChannelRequester::channelCreated(epics::pvData::Status const &Status,
                                       Channel::shared_pointer const &Channel) {
   // Seems that channel creation is actually a synchronous operation
   // and that this requester callback is called from the same stack
   // from which the channel creation was initiated.
-  LOG(Sev::Debug, "ChannelRequester::channelCreated:  (int)status.isOK(): {}",
-      (int)Status.isOK());
+  Logger->debug("ChannelRequester::channelCreated:  (int)status.isOK(): {}",
```
Comment (mattclarke) [inline src/EpicsClient/EpicsClientMonitor.cpp]: I think this whole method can be simplified to something like:
`std::string ChannelName = "Unknown";
  // May be possible to get name even if channel not created successfully.
  if (Channel) {
    ChannelName = Channel->getChannelName();
  }

  if (Status.isSuccess()) {
    Logger->debug("ChannelRequester::channelCreated: {}",
        ChannelName);
  } else {
    Logger->error("ChannelRequester::channelCreated:  failure: {}", ChannelName);
  }`

Comment (rerpha) [reply src/EpicsClient/EpicsClientMonitor.cpp]: @mattclarke surely doing `if (Channel)` is an implicit cast to bool? should it not be `if (Channel != nullptr)`?

File: src/EpicsClient/EpicsClientFactory.cpp [hunk 3]
```diff
@@ -17,29 +17,29 @@ std::unique_ptr<EpicsClientFactoryInit> EpicsClientFactoryInit::factory_init() {
...
-    LOG(Sev::Error,
+    Logger->error(
         "Reference count {} is not consistent, should never happen, but "
         "ignoring for now.",
         c);
     c = 0;
   }
   if (c == 0) {
-    LOG(Sev::Debug, "STOP   Epics factories");
+    Logger->debug("STOP   Epics factories");
```
Comment (rerpha) [inline src/EpicsClient/EpicsClientFactory.cpp]: I think this should be info, like the "start epics factories" statement

**Comment by rerpha** [comment]
It doesn't appear that the log files are being populated from the system tests - see the jenkins artifacts on the last PR build. 

worth checking this on the filewriter too as those log files have helped us identify bugs in the past.

**Comment by matthew-d-jones** [comment]
> It doesn't appear that the log files are being populated from the system tests - see the jenkins artifacts on the last PR build. Perhaps they are not being flushed/closed properly when the forwarder is exited?
> 
> worth checking this on the filewriter too as those log files have helped us identify bugs in the past.
> 
> we should use a flushing policy based on time:
> 
> https://github.com/gabime/spdlog/wiki/7.-Flush-policy

The problem was just that the verbosity was set low in the config files.

**Pull request merged**